### ISSUE TITLE
Speed up and refactor CI/CD workflows

### ISSUE THREAD

**Issue by ignacio-penas** [nuvlaedge/system-manager#51]
- [x] Make use of  build-push-actions, qemu, and buildx to build images in parallel and accelerate the build process.
- [x] Use cache in docker base images, python (for unittests) and intermediate builds
- [x] Create and independent workflow for development (nuvladev) images
~- [ ] #53~ descoped
